pagopa/io-app
# fix: [IA-496] If a payment started from a message detail, the cancellation of the payment causes a navigation to the Wallet instead of returning to the message details

## Short description
This pr fixes a minor glitch that caused a navigation to the Wallet instead of returning to the message details, if a payment started from a message CTA is canceled from the Payment Transaction Summary Screen.

https://user-images.githubusercontent.com/26501317/143471847-6bceeaee-5fbe-4edc-b01d-065da78b216c.mov


## List of changes proposed in this pull request
- Added a missing `dispatch` for the action `paymentInitializeState` in `PaymentButton`
- Renamed `deprecatedGetCurrentRouteName` and `deprecatedGetCurrentRouteKey` to `utilsGetCurrentRouteName` and `utilsGetCurrentRouteKey` in `ts/navigation/NavigationService.ts` (wrong naming)
- Changed `TransactionSummaryScreen` navigation params and added the new field `startRoute: NavigationLeafRoute | undefined;`. This was considered the best solution that allows not to radically change the payment flow, which is not totally atomic. A refactoring of this flow is always recommended, in order to make it atomic and lighten the strong dependencies it currently has.
- Added some new e2e tests `ts/__e2e__/payment.e2e.ts` in order to freeze this behaviour 

## How to test
**Manual testing**
- Messages > Tap a message with Payment > Show Notice > Cancel > Return to Message Detail

**Automatic e2e testing**
- Make sure you have the dev server up & running, with the `scripts/api-config.json` configuration.
- `yarn detox build --configuration ios.sim.debug;yarn detox test --configuration ios.sim.debug ts/__e2e__/payment.e2e.ts`

**CI e2e execution**
- https://app.circleci.com/pipelines/github/pagopa/io-app/16551/workflows/32b4c033-e9ec-4c89-8d0e-4f0b71c182f9/jobs/100592/artifacts
